### PR TITLE
ONYX-10850 Integrate mapping into validation process

### PR DESCRIPTION
**Pull request by sashaCher**
### What does this PR do?

Integrates mapping claims into validation process.
Automates basic sanity mapping test (ONYX-10472, ONYX-10473)

### What ticket does this PR close?
ONYX-10850

#### Change log
- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [X] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [X] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      annotations:
+        authn-jwt/raw/project-id: myproject
+        authn-jwt/raw/branch: main
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/hosts
+      member: !host myapp
+    """
+    And I add the secret value "branch:ref" to the resource "cucumber:variable:conjur/authn-jwt/raw/mapping-claims"
+
```
Comment (tzheleznyak) [inline cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:46]: Empty line

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    And I issue a JWT token:
+    """
+    {
+      "user":"myapp",
```
Comment (tzheleznyak) [inline cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:50]: in examples and annotations just use word valid and invalid to make it easier

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      id: myapp
+      annotations:
+        authn-jwt/raw/project-id: myproject
+        authn-jwt/raw/branch: main
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/hosts
+      member: !host myapp
+    """
+    And I add the secret value "branch:ref" to the resource "cucumber:variable:conjur/authn-jwt/raw/mapping-claims"
```
Comment (tzheleznyak) [inline cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:45]: We have more readable step for vairable set do this
`And I successfully set authn-jwt "mandatory-claims" variable to value "ref,sub"`

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        resource: !webservice
+    """
+    And I am the super-user
+    And I successfully set authn-jwt jwks-uri variable with value of "myJWKs.json" endpoint
+    And I successfully set authn-jwt "token-app-property" variable to value "user"
+
+  Scenario: ONYX-10473 Mapping claims with subsequent annotation
+    Given I extend the policy with:
+    """
+    - !variable conjur/authn-jwt/raw/mapping-claims
```
Comment (tzheleznyak) [inline cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:34]: add blank line between variable to host for readbility

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature (hunk 1 above)
Comment (sashaCher) [reply cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:46]: Removed

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature (hunk 2 above)
Comment (sashaCher) [reply cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:50]: Discussed it
Each field should have its own value.
For example `valid-project-id`.
Using the same value like `valid` for each fields can hide some problems

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature (hunk 3 above)
Comment (sashaCher) [reply cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:45]: Done

File: cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature (hunk 4 above)
Comment (sashaCher) [reply cucumber/authenticators_jwt/features/authn_jwt_mapping_claims.feature:34]: Done

**Review by tzheleznyak — APPROVED** [review]
LGTM

**Pull request merged**